src/catkin_pkg/topological_order.py

```diff
@@ -85,7 +86,22 @@ def calculate_depends_for_topological_order(self, packages):
                 # avoid function call to improve performance
                 # check within the loop since the set changes every cycle
                 continue
-            packages[name]._add_recursive_run_depends(packages, self.depends_for_topological_order)
+            packages[name]._add_recursive_run_depends_cached(packages, self.depends_for_topological_order)
+
+    def _add_recursive_run_depends_cached(self, packages, depends_for_topological_order):
+        """
+        Modify depends_for_topological_order argument by adding run_depends of self recursively.
+
+        This method is the cached variant of _add_recursive_run_depends().
+        Only packages which are in the passed packages are added and recursed into.
+
+        :param packages: dict of name to ``_PackageDecorator``
+        :param depends_for_topological_order: set to be extended
+        """
+        if self.run_depends_for_topological_order is None:
+            self.run_depends_for_topological_order = set()
+            self._add_recursive_run_depends(packages, self.run_depends_for_topological_order)
```

Comment:
Atm it seems that `run_depends_for_topological_order` contains its own package name which I don't think is desired (since a package is not allowed to depend on itself).

Reply:
Yes that was because the method added its own package first. I've changed it now to store the own package name not in the cache variable, and add the own package name to the given dependencies explicitly.

Reply:
After reviewing the patch again I notices that `depends_for_topological_order` also contains its own package name (which wasn't obvious from the comments / docblock). So I guess it is just consequential that `_recursive_run_depends_for_topological_order` can do the same.

I modified the patch to keep `self.package.name` in the cached member as you had it before but also added / updated the comments / docblock to reflect this detail: 6b2ac587aa437c16ed5969b7eb68c6d2b10274cb.